src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -338,12 +325,133 @@ public void storeContext(Context context, HttpServletRequest request) {
 		// Fill context with the required lti values.
 		// The VelocityViewServlet will take care of sending the processing on
 		// to the proper velocity template.
-		fillCcmValuesForContext(ltiValues, request);
+		fillCcmValuesForContext(ltiValues, request,tc);
 
 		context.put("ltiValues", ltiValues);
 	}
 
-	private void fillCcmValuesForContext(Map<String, String> ltiValues, HttpServletRequest request) {
+	private boolean isAccountAdmin(HttpServletRequest request) {
+		M_log.debug("isAdmin() call");
+		String courseId = request.getParameter(CUSTOM_CANVAS_COURSE_ID);
+		int userId = Integer.parseInt(request.getParameter(CUSTOM_CANVAS_USER_ID));
+
+		int courseAccountId= getCourseAccountId(courseId);
+		if(courseAccountId == 0){
+			return false;
+		}
+		M_log.debug("************** " + courseAccountId);
+		return determineIfUserIsAdminInHierarchy(userId,courseAccountId);
+	}
+
+	private boolean determineIfUserIsAdminInHierarchy(int userCanvasId, int courseAccountId){
+		M_log.debug("determineIfUserIsAdminInHierarchy() call");
+		boolean isAdmin = false;
+		String url = canvasURL + "/api/v1/accounts/" + courseAccountId + "/admins/?user_id[]=" + userCanvasId;
+		M_log.info("API call "+url);
+		HttpResponse response = processApiCall(new HttpGet(url));
+		if (response == null) {
+			M_log.error(String.format("check user %s is an admin in the course account %s had null response"
+					, userCanvasId, courseAccountId));
+			return isAdmin;
```

Comment:
Since computing the truth of isAdmin is easy in each "if" clause it is clearer to just return true or false directly.